### PR TITLE
Add unit test for quota config.

### PR DESCRIPTION
**Pull request by chowchow316**

### CODE REVIEW

File: contrib/endpoints/src/api_manager/config_test.cc [hunk 1]
```diff
@@ -870,6 +870,86 @@ TEST(Config, TestCorsDisabled) {
   ASSERT_EQ(nullptr, method1);
 }
 
+TEST(Config, TestInvalidMetricRules) {
+  MockApiManagerEnvironmentWithLog env;
+  static const char config_text[] = R"(
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/config_test.cc]: Please add comment to state why this is invalid
// There is no http.rule or api.method to match the selector.

File: contrib/endpoints/src/api_manager/config_test.cc [hunk 2]
```diff
@@ -870,6 +870,86 @@ TEST(Config, TestCorsDisabled) {
...
+  std::unique_ptr<Config> config = Config::Create(&env, config_text, "");
+  ASSERT_TRUE(config);
+
+  auto method1 = config->GetMethodInfo("GET", "/shelves");
+  ASSERT_NE(nullptr, method1);
+
+  auto metric_cost_vector = method1->metric_cost_vector();
+  ASSERT_EQ(2, metric_cost_vector.size());
+  ASSERT_EQ("test.googleapis.com/operation/request",
+            metric_cost_vector[0].first);
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/config_test.cc]: the order is un-determined. this test may fail in some platforms.

File: contrib/endpoints/src/api_manager/config_test.cc (hunk 1 above)
Comment (chowchow316) [inline contrib/endpoints/src/api_manager/config_test.cc]: Done.

File: contrib/endpoints/src/api_manager/config_test.cc (hunk 2 above)
Comment (chowchow316) [inline contrib/endpoints/src/api_manager/config_test.cc]: Removed the second <key, value> pair.

File: contrib/endpoints/src/api_manager/config_test.cc [hunk 3]
```diff
@@ -932,13 +929,10 @@ quota {
   ASSERT_NE(nullptr, method1);
 
   auto metric_cost_vector = method1->metric_cost_vector();
-  ASSERT_EQ(2, metric_cost_vector.size());
-  ASSERT_EQ("test.googleapis.com/operation/request",
-            metric_cost_vector[0].first);
-  ASSERT_EQ(10, metric_cost_vector[0].second);
+  ASSERT_EQ(1, metric_cost_vector.size());
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/config_test.cc]: but we DO want to test 2 metrics case.   You can either convert the vector to a "set", then compare the "set".  or sort the vector first,

File: contrib/endpoints/src/api_manager/config_test.cc [hunk 4]
```diff
@@ -870,6 +870,80 @@ TEST(Config, TestCorsDisabled) {
...
+         value: 200
+      }
+   }
+}
+)";
+
+  std::unique_ptr<Config> config = Config::Create(&env, config_text, "");
+  ASSERT_TRUE(config);
+
+  auto method1 = config->GetMethodInfo("GET", "/shelves");
```
Comment (rshriram) [inline contrib/endpoints/src/api_manager/config_test.cc]: To keep the code consistent with Envoy, I suggest avoiding over usage of auto method. In the Envoy codebase, we use explicit types most of the time

**Pull request merged**